packages/jest-test-sequencer/src/index.ts

```diff
@@ -65,6 +70,16 @@ export default class TestSequencer {
     return cache;
   }
 
+  shard(tests: Array<Test>, options: SortOptions): Array<Test> {
+    const shardSize = Math.ceil(tests.length / options.shardCount);
+    const shardStart = shardSize * (options.shardIndex - 1);
+    const shardEnd = shardSize * options.shardIndex;
+
+    return [...tests]
+      .sort((a, b) => (a.path > b.path ? 1 : -1))
```

Comment:
I wonder if we should be more clever here. Thoughts on calling `await this.sort(tests)` and then splitting off from there? and then using `index % options.shardIndex === 0` or something (not that exactly since it won't work 😅) since `sort` by default tries to schedule slowest/failing/largest first.

Might be overkill, dunno

Reply:
I've thought about this one a bit - my team will use a jump consistent hash (https://arxiv.org/pdf/1406.2294.pdf) based on the filename, sort based on that and then apply modulo. That should make for nicely distributed and very stable shards. 

In a later iteration I believe we could try and balance the shards to contain a similar amount of predicted test run time based on historical data, not sure yet what that would like yet though 😅

Reply:
TL;DR If you don't object I'll give a smarter shard algorithm a crack; thoughts on pulling in a jump consistent hashing lib as dependency vs. having it in source?

Reply:
Worth a go! Might be best to do that in a separate PR though, just to not block this one 🙂 I'm very interested in seeing that happen, though!

Reply:
I went ahead and added simple hashing of the test path, results are encouraging. The distribution of `e2e` test went from

```
108
102
0
0
```

to

```
51
48
62
49
```

Command used:

```
λ yarn jest --listTests --shard='1/4' | grep e2e | wc -l && yarn jest --listTests --shard='2/4' | grep e2e | wc -l && yarn jest --listTests --shard='3/4' | grep e2e | wc -l && yarn jest --listTests --shard='4/4' | grep e2e | wc -l
```

For CI this means:

<img width="1904" alt="Screen Shot 2022-03-06 at 12 02 11 am" src="https://user-images.githubusercontent.com/4248851/156884232-57b8049f-371e-4994-bc96-51a4cf116eb4.png">

Reply:
Oh yeah,  this is awesome! 8 minutes CI is a dream 😀